[CORE-8225] dt/archival: Turn off aws-chunked encoding when on GCS

wdberkeley opened the pull request:
boto3 by default will use request checksumming in some cases, like PUTs
to S3, which causes the content to be transferred with the aws-chunked
Content-Encoding. GCS will store the content verbatim and return it with
the same Content-Encoding, which breaks our code. This happens in the
inventory test when we use boto3 to upload faked inventory reports to
GCS using boto3 and GCS's S3-compatible-but-not-that-compatible API.
Redpanda fails to parse the aws-chunked encoding as JSON.

Fixes CORE-8225

## Backports Required

- [ ] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [X] none - papercut/not impactful enough to backport
- [ ] v25.3.x
- [ ] v25.2.x
- [ ] v25.1.x

## Release Notes

* none

wdberkeley commented:
/cdt
provider=gcp
tests/rptest/tests/cloud_storage_inventory_test.py

vbotbuildovich commented:
'cdt_instance_type' and 'region' is required if 'provider' is not 'aws'

[Workflow run logs][1].

[1]: https://github.com/redpanda-data/redpanda/actions/runs/19976156818

wdberkeley commented:
/cdt
provider=gcp
cdt_instance_type=
region=us-west2
cdt_instance_type=n2-highmem-4

wdberkeley commented:
/cdt
provider=gcp
region=us-west2
cdt_instance_type=n2-highmem-4
tests/rptest/tests/cloud_storage_inventory_test.py

wdberkeley commented:
/cdt
provider=aws
tests/rptest/tests/cloud_storage_inventory_test.py

wdberkeley committed:
dt/archival: Turn off aws-chunked encoding when on GCS

boto3 by default will use request checksumming in some cases, like PUTs
to S3, which causes the content to be transferred with the aws-chunked
Content-Encoding. GCS will store the content verbatim and return it with
the same Content-Encoding, which breaks our code. This happens in the
inventory test when we use boto3 to upload faked inventory reports to
GCS using boto3 and GCS's S3-compatible-but-not-that-compatible API.
Redpanda fails to parse the aws-chunked encoding as JSON.

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--5698acbd-1f1a-4f63-b0c0-e5d27b4257b5-->
<!--TEST RESULTS-->
<!--20361d30-242a-4af4-bf98-6b368bc4fc09-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/77553">build#77553</a></summary>


| test_class              | test_method           | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason   | test_history   |
|:------------------------|:----------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:---------|:---------------|
| ReplicatedMetastoreTest | TestBasicRemoveTopics |                  | unit        | https://buildkite.com/redpanda/redpanda/builds/77553#019b0081-5f03-4da4-a1d1-23ebc6705f47 | FAIL          | 0/1      |          |                |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/77588">build#77588</a></summary>


| test_class            | test_method                     | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                                                                                                                                                                           | test_history                                                                                                                                                        |
|:----------------------|:--------------------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| DataMigrationsApiTest | test_higher_level_migration_api | null             | integration | https://buildkite.com/redpanda/redpanda/builds/77588#019b042b-b97b-479e-b6d2-b21f74c4327d | FLAKY         | 8/11     | Test FAILS after retries.[Significant increase in flaky rate](https://redpandadata.atlassian.net/wiki/spaces/DEVPROD/pages/1419247661/Statistical+retry+failure+reasons#%5BinlineExtension%5DCase-B%3A-Test-fails-with-an-improbably-higher-rate-than-baseline-(run-fails))(baseline=0.0085, p0=0.0031, reject_threshold=0.0100) | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DataMigrationsApiTest&test_method=test_higher_level_migration_api |
</details>
<!--/body-->

wdberkeley commented:
/cdt
provider=gcp
region=us-west2
cdt_instance_type=n2-highmem-4
tests/rptest/tests/cloud_storage_inventory_test.py

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/77588-->
<!--124c128e-f4ca-42df-b58c-88cab78b445f-->
<!--title-->
## Retry command for [Build#77588](https://buildkite.com/redpanda/redpanda/builds/77588)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
skip-redpanda-build
skip-units
skip-rebase
tests/rptest/tests/data_migrations_api_test.py::DataMigrationsApiTest.test_higher_level_migration_api
```
<!--/body-->

wdberkeley commented:
/cdt
provider=gcp
region=us-west2
cdt_instance_type=n2-highmem-4
tests/rptest/tests/cloud_storage_inventory_test.py

wdberkeley commented:
/cdt
provider=aws
tests/rptest/tests/cloud_storage_inventory_test.py

wdberkeley commented:
Note to self: re-run the test on AWS and GCS before merging.

wdberkeley commented:
/cdt
provider=gcp
region=us-west2
cdt_instance_type=n2-highmem-4
tests/rptest/tests/cloud_storage_inventory_test.py

wdberkeley commented:
/cdt
provider=aws
tests/rptest/tests/cloud_storage_inventory_test.py

dotnwat reviewed (approved):
wow nice find. how did you discover it?

also

> Redpanda fails to parse the aws-chunked encoding as JSON

this is because we don't support chunked encoding, or some other reason?

wdberkeley commented:
> wow nice find. how did you discover it?

The json parser logged what it couldn't parse. I had AI look at it and it was like "that's aws-chunked encoding!".

> also
> 
> > Redpanda fails to parse the aws-chunked encoding as JSON
> 
> this is because we don't support chunked encoding, or some other reason?

We don't support encoding or decoding using aws-chunked. The aws-chunked encoding is meant for chunked uploads for S3 and isn't meant to be used to send data back to clients, so clients aren't meant to have to decode it. The problem is GCP doesn't handle aws-chunked the same way S3 does.
